### ISSUE TITLE
Ignore pinned tabs

### ISSUE THREAD

**Issue by pieter91** [Swift-open-source/UltraTabSaver#12]
Hi! Thanks so much for this extension. It's been really helpful so far. 

The only downside to the functionality I've found is that pinned tabs aren't treated any differently, meaning they are saved and closed along with all the regular tabs. When closing they also close on other opened windows. 

Would it be possible to (set the plugin to) ignore any pinned tabs? 

I went looking into this, but couldn't find any relevant documentation from Apple. Is there a way to detect wether a tab is pinned?

**Comment by morsamatias** [comment]
Hey, thank you for the feedback. You are right, I'm going to see if I found any way to don't save the pinned tabs.

PS: You can star the repo to see the updates.

I would like to know more opinions about other users to see if it is better to:
1) Save the pinned tabs
2) Don't save the pinned tabs
3) Make it configurable

**Comment by kevinpapst** [comment]
I just opened an issue about pinned tabs in the Session Buddy repo with the same problem.

My workflow (and I guess of many other users) is:
having a handful of pinned tabs which are necessary to my work as I need and use them all day long.

What I would like to see: a checkbox "Ignore pinned tabs".
When activated saving a session should not save them. 
When closing tabs the same applies: do not touch them.

So basically the checkbox means "I really know what I am doing here, please leave my pinned tabs alone".

This (at least for me) is a necessary feature that currently (and unfortunately) prevents me from using the extension.
I know, every user thinks his feature request is the most important one 😁 just wanted to leave a vote with an explanation why such a preference might be crucial to some users.

**Comment by morsamatias** [comment]
Hi @kevinpapst thank you for the feedback!. Where is the Session Buddy repo, can you share the link please?
Actually I had just an idea to fix them, but maybe it's more tedious, The user will have to add one by one the urls of the pinned tabs, and he can manage them from the menu, what do you think?

**Comment by kevinpapst** [comment]
@morsamatias I didn't wanted to post a link to a competitor, thats why I didn't add it initially. It is called "Session Pal" in the AppStore. Here it is: https://github.com/dwarvesf/session-buddy/ and [this](https://github.com/dwarvesf/session-buddy/issues/10) is my issue which I posted before I found your plugin.

I know it often looks easier from the outside, I was hoping (without knowing the available APIs) that there might be a simple solution like a bool flag `isPinned` on the tabs that you save. Then you could simply ignore each tab which has `true` as result.

Following a couple of discussions it seems there isn't. But it seems to be possible to detect with a workaround by checking the containingWindow for nil as discussed [here](https://developer.apple.com/forums/thread/123120) or [here](https://stackoverflow.com/questions/63509871/check-if-tab-page-is-pinned-in-safari-app-extension).
I have almost zero Swift knowledge, maybe there could be a check added [somewhere in saveAllPages()](https://github.com/Swift-open-source/UltraTabSaver/blob/main/Ultra%20TabSaver%20Extension/Persistance.swift#L135)?

**Comment by morsamatias** [comment]
@kevinpapst Have you used Session Pal? It's good?

You are right it looks like getContainingWindow is the answer, give me a couple of days and I'm going to add this feature. 

Thank you for the help, feel free to add more issues to keep improving this project!

**Comment by Tukan3** [comment]
I also would really like to see this feature added.
